code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -728,6 +727,44 @@ func (sc *StorageSmartContract) getAllocationBlobbers(alloc *StorageAllocation,
 	return
 }
 
+func AggregateErrorsFromChannel(ch chan error) []error {
+	errs := make([]error, 0)
+	for i := 0; i < len(ch); i++ {
+		select {
+		case err := <-ch:
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func GetAggregatedError(errs []error, sorted bool) error {
+	if len(errs) == 0 {
+		return nil
+	}
+
+	errStrings := make([]string, 0)
+	for _, err := range errs {
+		errStrings = append(errStrings, err.Error())
+	}
+
+	if sorted {
+		sort.Strings(errStrings)
+	}
+
+	err := errors.New(errStrings[0])
+	if len(errs) == 1 {
+		return err
+	}
```

Comment:
this can be moved to line 746 so that all the computation from lines 747-754 need not be done.